src/borg/platform/win32.py

```diff
@@ -0,0 +1,41 @@
+from functools import lru_cache
+
+@lru_cache(maxsize=None)
+def uid2user(uid, default=None):
+    print(uid, default)
+    try:
+        return pwd.getpwuid(uid).pw_name
+    except KeyError:
+        return default
+
+
+@lru_cache(maxsize=None)
+def user2uid(user, default=None):
+    print(user, default)
+    try:
+        return user and pwd.getpwnam(user).pw_uid
+    except KeyError:
+        return default
+
+
+@lru_cache(maxsize=None)
+def gid2group(gid, default=None):
+    print(gid, default)
+    try:
+        return grp.getgrgid(gid).gr_name
+    except KeyError:
+        return default
+
+
+@lru_cache(maxsize=None)
+def group2gid(group, default=None):
+    print(group, default)
+    try:
+        return group and grp.getgrnam(group).gr_gid
+    except KeyError:
+        return default
+
+def getosusername():
+    """Return the os user name."""
+    uid = os.getuid()
+    return uid2user(uid, uid)
```

Comment:
end all files with a single newline (there should be a setting in your editor for this), so that diff tools and github diff display do not complain.

Reply:
also, remove the leftover debug prints.

Reply:
Also, these 5 functions look identical / copy & pasted from posix.py.

If you don't need them to be different, just import them from there so no code is duplicated?

Reply:
This part is in progress and by no means complete or correctly working! Updates will follow.